vr4300/pipeline.c

```diff
@@ -361,6 +361,102 @@ static int vr4300_dc_stage(struct vr4300 *vr4300) {
       line->metadata |= exdc_latch->request.type;
     }
 
+    else if (unlikely(exdc_latch->request.type > VR4300_BUS_REQUEST_CACHE_WRITE)) {
+      // ll/lld/sc/scd
+      uint64_t dword, rtemp, wtemp, wdqm;
+      unsigned shiftamt, rshiftamt, lshiftamt;
+      uint32_t s_paddr;
+
+      if (exdc_latch->request.type == VR4300_BUS_REQUEST_READ_CONDITIONAL) {
+        vr4300->regs[VR4300_REGISTER_LLBIT] = 1;
+        vr4300->regs[VR4300_CP0_REGISTER_LLADDR] = paddr >> 4;
+      } else {
+        if (!vr4300->regs[VR4300_REGISTER_LLBIT] ||
+            vr4300->regs[VR4300_CP0_REGISTER_LLADDR] != paddr >> 4) {
+          // Fail the store, set the reg to 0, write nothing to memory
+          exdc_latch->request.type = VR4300_BUS_REQUEST_NONE;
+          vr4300->regs[exdc_latch->request.sc_reg] = 0;
+          return 0; // fail is not an exception
+        } else {
+          // Successful store, set the reg to 1
+          vr4300->regs[exdc_latch->request.sc_reg] = 1;
```

Comment:
Should this be a pipelined write?  Otherwise, the result may not be correctly forwarded:
https://github.com/n64dev/cen64/blob/master/vr4300/pipeline.c#L173
```
dcwb_latch->dest = exdc_latch->request.sc_reg;
dcwb_latch->result = 1
```